### PR TITLE
Api_register

### PR DESCRIPTION
**Pull request by MarioGafeed**

### CODE REVIEW

File: app/Http/Controllers/UsersController.php [hunk 1]
```diff
@@ -47,7 +47,8 @@ public function create()
      */
     public function store(UsersRequest $request)
     {
-        $user = $this->UserInterface->store($request->all());
+        $user = $this->UserInterface->store($request->all());
```
Comment (mo7zayed) [inline app/Http/Controllers/UsersController.php:50]: `userInterface` not `UserInterface`

File: app/Http/Requests/UsersRequest.php [hunk 2]
```diff
@@ -34,7 +34,7 @@ public function rules()
         ];
 
         if (in_array($this->method(), ['POST', 'PATCH'])) {
```
Comment (mo7zayed) [inline app/Http/Requests/UsersRequest.php:36]: Thy having `POST` to check updating the data?

File: app/Http/Controllers/UsersController.php (hunk 1 above)
Comment (MarioGafeed) [reply app/Http/Controllers/UsersController.php:50]: Done

File: app/Http/Requests/UsersRequest.php (hunk 2 above)
Comment (MarioGafeed) [reply app/Http/Requests/UsersRequest.php:36]: Done

**Pull request merged**